Add GSoC information pages (#873)
knative/community: google-summer-of-code/gsoc-2022.md

### Application period

If you are a contributor looking forward to participating in the GSoC 2022 with the Knative community, feel free to
browse the growing idea list below.

What should I do before to get accepted:

* Check out the program [website](https://summerofcode.withgoogle.com/).
* Understand how the program [works](https://summerofcode.withgoogle.com/how-it-works).
* Read the [contributor guide](https://google.github.io/gsocguides/student/) of the program.
* Make sure you are [eligible](https://summerofcode.withgoogle.com/terms/contributor) to participate in the program.
* Check out the [differences](https://opensource.googleblog.com/2021/11/expanding-google-summer-of-code-in-2022.html)
  in the program this year.
* Try Knative by going through the [getting started](https://knative.dev/docs/getting-started/) page.
* Interact with the community using the [communication channels](#communication).
* Browse the [idea list](#idea-list) below. The earlier, the better.
* Check out the Knative community [website](https://knative.dev/docs/community/).

Tips:

* Ask for help if you need assistance. The community will select the contributors who are not only good technically.

>   Community expects the contributors to have interaction.

Reviewer: ```suggestion
  The community expects the contributors to have interaction.
```

* The preferred way of asking for assistance is using the [communication channels](#communication). In specific cases,
  you can privately contact the mentor if you feel uncomfortable about the public communication.
* Start exploring early!
* Have good interaction, so that the community recognizes your name! Start by introducing yourself to the community.
* Be polite!

> * Help others! Help the like-minded potential contributors in the public channels.

Reviewer: ```suggestion
* Help others! Help like-minded potential contributors in the public channels.
```

* Check out the projects from [previous years](https://summerofcode.withgoogle.com/archive).
* For a sample project proposal document, check out the [ideas](#idea-list) section.

### Coding period

To be added later

## Mentors

### Application period

> Any Knative community member can submit ideas and volunteer to mentor students. However, Google gives limited number of

Reviewer: ```suggestion
Any Knative community member can submit ideas and volunteer to mentor students. However, Google gives a limited number of
```

> slots, so, community will have to select the best idea+contributor+mentor combination.

Reviewer: ```suggestion
slots, so, the community will have to select the best idea+contributor+mentor combination.
```

Do not hesitate to contact the [organization administrators](#organization-administrators) for any questions.

What should I do in the application period?

* First of all, make sure you are aware of the expectations from you (see below).
* Check out the program [website](https://summerofcode.withgoogle.com/).
* Understand how the program [works](https://summerofcode.withgoogle.com/how-it-works).
* Read the [mentor guide](https://google.github.io/gsocguides/mentor/) of the program.
* Check out the [differences](https://opensource.googleblog.com/2021/11/expanding-google-summer-of-code-in-2022.html)
  in the program this year, if you have experience with GSoC before.
* Add your idea to the [idea list](#idea-list) below. The earlier, the better.
* Monitor the [communication channels](#communication) for potential contributor questions.

Expectations from mentors:

* Add your ideas [below](#idea-list).
* Make sure you have the capacity to mentor a contributor. It is not the entire Knative community mentoring a
  contributor; it is YOU mentoring the contributor. This depends on the complexity of the project and the technical

>   knowledge of the contributor, but as a general rule, expect spending 4 hours a week.

Reviewer: ```suggestion
  knowledge of the contributor, but expect to spend 4 hours a week as a general rule.
```

* Help the contributor to not only finish their projects but also teach about matters that are part of an open source
  community such as how project governance is, how communication in the community works and similar.
* Read the [mentor guide](https://google.github.io/gsocguides/mentor/) of the program.
* Do NOT share any private information about the contributor candidates publicly, such as if you would like to mentor a

>   specific candidate or if you think candidate is not good enough for the program. Reach out to the administrators for

Reviewer: ```suggestion
  specific candidate or if you think the candidate is not good enough for the program. Reach out to the administrators for
```

  that.

Tips:

> * Be polite and understanding! Not everyone has your knowledge on the technology we are working on at Knative community.

Reviewer: ```suggestion
* Be polite and understanding! Not everyone has your knowledge of the technology we are working on at Knative community.
```

* Be professional! Don't forget that the contributors could be part of the community one day.

> * Send a link of this page to the contributors when you think they can benefit the tips written here.

Reviewer: ```suggestion
* Send a link to this page to the contributors when you think they can benefit from the tips written here.
```

* Do not hesitate to contact the [administrators](#organization-administrators).

### Coding period

To be added later

## Communication

> Contributors are welcome to participate any talks and discussions in the Knative community channels.

Reviewer: ```suggestion
Contributors are welcome to participate in any talks and discussions in the Knative community channels.
```

* Knative users group: https://groups.google.com/g/knative-users
    * Preferred communication channel. Introduce yourself here as a candidate contributor.

> * Slack: https://slack.knative.dev/ Explore the channels yourselves, however make sure you're checking these:

Reviewer: ```suggestion
* Slack: https://slack.knative.dev/ Explore the channels yourselves, however, make sure you're checking these:
```
